pkg/apimachinery/identity/wrapper.go

```diff
@@ -13,6 +14,37 @@ type IDClaimsWrapper struct {
 	Source Requester
 }
 
+func (i *IDClaimsWrapper) NamespaceMatches(namespace string) bool {
+	parts := strings.Split(i.Source.GetAllowedKubernetesNamespace(), "-")
+	// for multi-tenant operators, access tokens will have a namespace claim of "*"
+	if parts[0] == "*" {
+		return true
+	}
+
+	if len(parts) < 2 {
+		return false
+	}
+
+	// for cluster-scoped resources, namespace in request is ""
```

Comment:
i'll need to check how this is used, but is there a chance that this could accidentally allow listing across namespaces for namespaced resources?

Reply:
It won't unless the namespace claim is set to "*" in the JWT. Does that make sense?

Reply:
But since I do `parts[0] == "*"` earlier and return true unconditionally, I think you are right. I need to return false here.

Reply:
I am thinking of either having `NamespaceMatches` being completely removed from `AccessClaims` and `IdentityClaims` in favor of another `NamespaceAccessChecker` interface also coming in from authlib.

In any case, depending on an empty namespace value is not intuitive to signal cluster-scoped resources for me either. So I will change it to explicitly passing in `*` when asking that question of a `NamespaceMatches` or the access checker.